components/eamxx/src/physics/mam/eamxx_mam_wetscav_process_interface.cpp

```diff
@@ -0,0 +1,549 @@
+        wet_aero_.cld_aero_mmr[imode][ispec] =
+            get_field_out(cld_mmr_field_name).get_view<Real **>();
+        dry_aero_.cld_aero_mmr[imode][ispec] =
+            buffer_.dry_cld_aero_mmr[imode][ispec];
+      }
+    }
+  }
+
+  //---------------------------------------------------------------------------------
+  // Allocate memory
+  //---------------------------------------------------------------------------------
+  // Alllocate aerosol-related gas tendencies
+  for(int g = 0; g < mam_coupling::num_aero_gases(); ++g) {
+    dry_aero_tends_.gas_mmr[g] = view_2d("gas_mmr", ncol_, nlev_);
+  }
+
+  // Allocate aerosol state tendencies (interstitial aerosols only)
+  for(int imode = 0; imode < mam_coupling::num_aero_modes(); ++imode) {
+    dry_aero_tends_.int_aero_nmr[imode] = view_2d("int_aero_nmr", ncol_, nlev_);
+
+    for(int ispec = 0; ispec < mam_coupling::num_aero_species(); ++ispec) {
+      dry_aero_tends_.int_aero_mmr[imode][ispec] =
+          view_2d("int_aero_mmr", ncol_, nlev_);
+    }
+  }
+
+  // Allocate work array
+  const int work_len = mam4::wetdep::get_aero_model_wetdep_work_len();
+  work_              = view_2d("work", ncol_, work_len);
+
+  //---------------------------------------------------------------------------------
+  // Setup preprocessing and post processing
+  //---------------------------------------------------------------------------------
+  // set up our preprocess  and postprocess functors
+  preprocess_.initialize(ncol_, nlev_, wet_atm_, wet_aero_, dry_atm_,
+                         dry_aero_);
+
+  postprocess_.initialize(ncol_, nlev_, wet_atm_, wet_aero_, dry_atm_,
+                          dry_aero_);
+}
+
+// ================================================================
+//  RUN_IMPL
+// ================================================================
+void MAMWetscav::run_impl(const double dt) {
+  const auto scan_policy = ekat::ExeSpaceUtils<
+      KT::ExeSpace>::get_thread_range_parallel_scan_team_policy(ncol_, nlev_);
+
+  // preprocess input -- needs a scan for the calculation of all variables
+  // needed by this process or setting up MAM4xx classes and their objects
+  Kokkos::parallel_for("preprocess", scan_policy, preprocess_);
+  Kokkos::fence();
+
+  const mam_coupling::DryAtmosphere &dry_atm = dry_atm_;
+  const auto &dry_aero                       = dry_aero_;
+  const auto &work                           = work_;
+  const auto &dry_aero_tends                 = dry_aero_tends_;
+
+  // ---------------------------------------------------------------
+  // These variables are "required" or pure inputs for the process
+  // ---------------------------------------------------------------
+
+  //----------- Variables from convective scheme -------------
+
+  // TODO: Following variables are from convective parameterization (not
+  // implemented yet in EAMxx), so should be zero for now
+
+  auto sh_frac = view_2d("sh_frac", ncol_, nlev_);
```

Comment:
I believe the issue here is that we are allocating memory at every time step.

The solution is to create a class member:

In the .hpp file:

`view_2d sh_frac_;`

In run_iml:

`sh_frac_ = view_2d("sh_frac", ncol_, nlev_);
`
@singhbalwinder ,I can make this change if you want.

Reply:
Yes, that is correct. These variables will have valid values when low-res EAMxx is ready. For now, I agree we should pre-allocate these variables in the `initialize_impl` to avoid the cost of allocating them at each time step. I have asked for @overfelt's help to move these allocations out of `run_impl`.